diff --git a/lib/setup.php b/lib/setup.php
index adc3c47..61ffcfb 100644
--- a/lib/setup.php
+++ b/lib/setup.php
@@ -793,6 +793,13 @@ if (!empty($CFG->allowthemechangeonurl) and !empty($_GET['theme'])) {
 }
 unset($urlthemename);
 
+// Begin TDM: process theme change in co-branding
+require_once "{$CFG->dirroot}/local/tdmcobrand/lib.php";
+try {
+    local_tdmcobrand_rule::setup_theme();
+} catch (Exception $e) { /* the site probably isn't installed yet */ }
+// End TDM
+
 // Ensure a valid theme is set.
 if (!isset($CFG->theme)) {
     $CFG->theme = 'standardwhite';
